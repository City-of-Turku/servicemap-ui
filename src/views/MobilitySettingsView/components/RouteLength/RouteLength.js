import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const RouteLength = ({ classes, intl, route }) => {
  const formatRoutelength = inputLength => Math.round(inputLength / 1000);

  const renderRouteText = (routeName) => {
    switch (routeName) {
      case 'EuroVelo':
        return (
          <Typography
            component="p"
            variant="body2"
            aria-label={intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.euroVelo' })}
            className={classes.margin}
          >
            {intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.euroVelo' })}
          </Typography>
        );
      case 'Saariston rengastie':
        return (
          <Typography
            component="p"
            variant="body2"
            aria-label={intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.archipelagoTrail' })}
            className={classes.margin}
          >
            {intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.archipelagoTrail' })}
          </Typography>
        );
      case 'Aurajoentie':
        return (
          <Typography
            component="p"
            variant="body2"
            aria-label={intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.auraRiverTrail' })}
            className={classes.margin}
          >
            {intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.auraRiverTrail' })}
          </Typography>
        );
      default:
        return null;
    }
  };

  const generateTranslations = (routeName) => {
    const split = routeName.split(' ');
    const [a, b] = split;
    if (a === 'Seutureitti') {
      return (
        <Typography
          component="p"
          variant="body2"
          aria-label={intl.formatMessage({ id: `mobilityPlatform.menu.bicycleRoutes.regionalTrail${b}` })}
          className={classes.margin}
        >
          {intl.formatMessage({ id: `mobilityPlatform.menu.bicycleRoutes.regionalTrail${b}` })}
        </Typography>
      );
    }
    return renderRouteText(routeName);
  };

  return (
    <div className={classes.container}>
      <div className={classes.paragraph}>
        <Typography
          component="p"
          variant="body2"
          aria-label={`${intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.length' })} ${formatRoutelength(
            route.length,
          )} km.`}
        >
          {intl.formatMessage({ id: 'mobilityPlatform.menu.bicycleRoutes.length' })}
          {' '}
          {formatRoutelength(route.length)}
          {' '}
          km.
        </Typography>
        {generateTranslations(route.name_fi)}
      </div>
    </div>
  );
};

RouteLength.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  route: PropTypes.objectOf(PropTypes.any),
};

RouteLength.defaultProps = {
  route: null,
};

export default RouteLength;
