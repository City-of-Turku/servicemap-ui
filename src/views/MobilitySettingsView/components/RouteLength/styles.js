const styles = theme => ({
  paragraph: {
    textAlign: 'left',
    padding: theme.spacing(1.5),
    width: '90%',
  },
  margin: {
    marginTop: theme.spacing(1),
  },
});

export default styles;
