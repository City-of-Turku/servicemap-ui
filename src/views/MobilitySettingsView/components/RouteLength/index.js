import RouteLength from './RouteLength';

export default RouteLength;
