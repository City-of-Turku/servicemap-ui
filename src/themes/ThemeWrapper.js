import { ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';
import themes from '.';
import config from '../../config';
import { selectThemeMode } from '../redux/selectors/user';

// If external theme (by Turku) is true, then can be used to select which theme to get
const externalTheme = config.themePKG;
const isExternalTheme = !externalTheme || externalTheme === 'undefined' ? null : externalTheme;

/**
 * Returns default theme based on env value
 * @returns theme
 */
const getTheme = () => {
  if (isExternalTheme) {
    return themes.SMThemeTku;
  }
  return themes.SMTheme;
};

// Component to handle theme changes
const ThemeHandler = ({ children }) => {
  const themeMode = useSelector(selectThemeMode);
  return (// Get correct theme setting from store
    <ThemeProvider theme={themeMode === 'dark' ? themes.SMThemeDark : getTheme()}>
      {children}
    </ThemeProvider>
  );
};
ThemeHandler.propTypes = {
  children: PropTypes.node,
};
ThemeHandler.defaultProps = {
  children: null,
};

export default ThemeHandler;
