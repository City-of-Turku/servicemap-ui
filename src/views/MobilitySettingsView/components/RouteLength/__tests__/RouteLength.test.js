// Link.react.test.js
import React from 'react';
import { ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import themes from '../../../../../themes';
import RouteLength from '../index';
import { initialState } from '../../../../../redux/reducers/user';
import finnishTranslations from '../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  route: {
    name_fi: 'EuroVelo',
    length: 100000,
  },
};

const mockStore = configureStore([]);

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => {
  const store = mockStore({
    user: initialState,
    settings: {},
  });

  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
};

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<RouteLength />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<RouteLength {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<RouteLength {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual('Reitin pituus: 100 km.');
    expect(p[1]).toBeInTheDocument();
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<RouteLength {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toEqual('Reitin pituus: 100 km.');
    expect(p[1].getAttribute('aria-label')).toBeTruthy();
  });
});
