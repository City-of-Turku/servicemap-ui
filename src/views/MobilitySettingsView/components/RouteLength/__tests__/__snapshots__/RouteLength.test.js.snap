// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteLength /> should work 1`] = `
<div>
  <div>
    <div
      class="injectIntl(RouteLength)-paragraph-1"
    >
      <p
        aria-label="Reitin pituus: 100 km."
        class="MuiTypography-root MuiTypography-body2"
      >
        Reitin pituus:
         
        100
         
        km.
      </p>
      <p
        aria-label="EuroVelo 10 on eurooppalainen Suomen rannikkoa seuraava polkupyöräreitti. Helsingin ja Turun välisellä matkalla reitti on merkitty opastein."
        class="MuiTypography-root injectIntl(RouteLength)-margin-2 MuiTypography-body2"
      >
        EuroVelo 10 on eurooppalainen Suomen rannikkoa seuraava polkupyöräreitti. Helsingin ja Turun välisellä matkalla reitti on merkitty opastein.
      </p>
    </div>
  </div>
</div>
`;
